Jenkinsfile

```diff
@@ -48,21 +47,21 @@ for(j = 0; j < jdks.size(); j++) {
                 // Now run the actual build.
                 stage("${buildType} Build / Test") {
                     timeout(time: 300, unit: 'MINUTES') {
+                      realtimeJUnit(healthScaleFactor: 20.0, testResults: '*/target/surefire-reports/*.xml,war/junit.xml') {
                         // -Dmaven.repo.local=… tells Maven to create a subdir in the temporary directory for the local Maven repository
                         // -ntp requires Maven >= 3.6.1
-                        def mvnCmd = "mvn -Pdebug -Pjapicmp -U -Dset.changelist help:evaluate -Dexpression=changelist -Doutput=$changelistF clean install ${runTests ? '-Dmaven.test.failure.ignore' : '-DskipTests'} -V -B -ntp -Dmaven.repo.local=$m2repo -Dspotbugs.failOnError=false -Dcheckstyle.failOnViolation=false -e"
+                        def mvnCmd = "mvn -Pdebug -Pjapicmp -U -Dset.changelist help:evaluate -Dexpression=changelist -Doutput=$changelistF clean install -Dmaven.test.failure.ignore -V -B -ntp -Dmaven.repo.local=$m2repo -Dspotbugs.failOnError=false -Dcheckstyle.failOnViolation=false -e"
                         infra.runWithMaven(mvnCmd, jdk.toString(), javaOpts, true)
 
                         if(isUnix()) {
                             sh 'git add . && git diff --exit-code HEAD'
                         }
+                      }
                     }
                 }
 
                 // Once we've built, archive the artifacts and the test results.
                 stage("${buildType} Publishing") {
-                    if (runTests) {
-                        junit healthScaleFactor: 20.0, testResults: '*/target/surefire-reports/*.xml,war/junit.xml'
```

Comment:
bit concerned that this will break checks api integration which is really nice:
https://github.com/jenkinsci/junit-plugin/issues/320

Reply:
So what is the next step Tim? Close this PR?

Reply:
I'll take a look at the junit plugin in between form modernisation work and see how hard it would be to just properly integrate it or patch real time junit

Reply:
I think this PR would likely integrate it already: https://github.com/jenkinsci/junit-realtime-test-reporter-plugin/pull/12

Reply:
I’m doing some testing on it and should have a release out by tomorrow so if we can just hold off till then that would be great (just want to make sure it’s viable)